node/task.ts

```diff
@@ -516,15 +516,22 @@ export function getEndpointUrl(id: string, optional: boolean): string {
  * @param id name of the service endpoint
  * @param key of the parameter
  * @param optional whether the endpoint data is optional
+ * @param secret, optional, if set to true, will store the value as a secret variable so it will be masked in the build logs
  * @returns {string} value of the endpoint data parameter
  */
-export function getEndpointDataParameter(id: string, key: string, optional: boolean): string {
-    var dataParamVal = process.env['ENDPOINT_DATA_' + id + '_' + key.toUpperCase()];
+export function getEndpointDataParameter(id: string, key: string, optional: boolean, secret?: boolean): string {
+    var dataVariable = 'ENDPOINT_DATA_' + id + '_' + key.toUpperCase();
+    var dataParamVal = process.env[dataVariable];
 
     if(!optional && !dataParamVal) {
         throw new Error(loc('LIB_EndpointDataNotExist', id, key));
     }
 
+    if(secret) {
+        setVariable(dataVariable, dataParamVal, true);
+        dataParamVal = vault.retrieveSecret('SECRET_' + dataVariable);
+    }
+
```

Comment:
// for RC2.  In RTM after 2.x has use command setSecret
if (secret) {
   debug(id + ' data ' + key + ' hasValue ' + dataParamVal ? true : false);
}
else {
    debug(id + ' data ' + key + ' = ' + dataParamVal);
}